Add controller for MksPipelineRun

UtkarshChaurasia opened the pull request:
### Add controller for MksPipelineRun

- Listens for MksPipelineRun resource being created, updated, or deleted.
- Adds corresponding TaskPipeline using Tekton API.

### How to test

- Build and run the executable using steps in README.md.
- Apply the CRDs and examples in the config folder.
- Add a new MksPipelineRun resource and see for logs.

### Related issues

Fixes #12

UtkarshChaurasia commented:
Hello @piyush-garg, please review.

avinal commented on .gitignore:
```diff
@@ -18,3 +18,4 @@ vendor/
 
 .vscode
 .idea
+
```
```suggestion
```
Remove newline changes (unnecessary)

avinal commented on main.go:
```diff
@@ -1,9 +1,90 @@
+		klog.Fatalf("Error building example clientset: %v", err)
+	}
+
+	/*deployment := &v1alpha1.MksPipelineRun{
+		TypeMeta:   v1.TypeMeta{Kind: "MksPipeline"},
+		ObjectMeta: v1.ObjectMeta{Name: "hello10"},
+		Spec:       v1alpha1.MksPipelineRunSpec{PipelineRef: v1alpha1.MksPipelineRunRef{Name: "hello"}},
+	}
+
+	crt, err := exampleClient.MkscontrollerV1alpha1().MksPipelineRuns("default").Create(context.TODO(), deployment, v1.CreateOptions{})
+	if err != nil {
+		klog.Fatalf("Error creating all resources: %v", err)
+	}
+	fmt.Println(crt)
+
+	list, err := exampleClient.MkscontrollerV1alpha1().MksPipelineRuns("default").Get(context.TODO(), "hello10", v1.GetOptions{})
+	if err != nil {
+		klog.Fatalf("Error listing all databases: %v", err)
+	}
+	fmt.Println(list)
+
+	gt, err := exampleClient.MkscontrollerV1alpha1().MksPipelineRuns("default").List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		klog.Fatalf("Error listing all databases: %v", err)
+	}
+	for _, name := range gt.Items {
+		fmt.Println(name)
+
+	}*/
```
```suggestion
```

avinal commented on main.go:
```diff
@@ -1,9 +1,90 @@
+	informers "github.com/MiniTeks/mks-server/pkg/client/informers/externalversions"
+	con "github.com/MiniTeks/mks-server/pkg/controllers/mkspipelinerun"
+
+	//v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
```suggestion
```

avinal commented on main.go:
```diff
@@ -1,9 +1,90 @@
 	"fmt"
+	"time"
+
+	//"github.com/MiniTeks/mks-server/pkg/apis/mkscontroller/v1alpha1"
```
```suggestion
```

avinal commented on main.go:
```diff
@@ -1,9 +1,90 @@
 package main
 
 import (
+	//"context"
```
```suggestion
```

avinal commented:
Please remove commented code and sqash all commits to at most 2 commits.

UtkarshChaurasia commented:
> Please remove commented code and sqash all commits to at most 2 commits.

Thank you. Did the suggested changes

UtkarshChaurasia committed:
Implemeted MksPipelineRun controller

- Use Tekton APIs
- Add handlers to convert the mksPipelineRun to Tekton pipelinerun when a new
  resorce is added.
